### PR TITLE
cigar manipulation function

### PR DESCRIPTION
**Pull request by dthybert**
## Description

This is a same than for the #420. 

When trying solving travis issue with Thiago we merged by error. So this is a second PR taking into account Botond suggestions which was typo correction.

### CODE REVIEW

**Review by ens-sb — APPROVED** [review]

**Review by twalsh-ebi — COMMENTED** [review]
I've added some comments/suggestions here and some in PR #420.

**Review by twalsh-ebi — CHANGES_REQUESTED** [review]
This is just a quick first pass on the revised changes. I will follow up with a full review.

**Review by twalsh-ebi — COMMENTED** [review]
It's looking well, and I've just added a last few minor comments.

Though not apparent in #435, I noticed a typo in #420, which can be seen in the license text on [line 5 of the CIGAR module](https://github.com/Ensembl/ensembl-compara/blob/ae7c5ad9c0d9d9fc7c7db82b2c2d792e0eda830d/src/python/lib/ensembl/compara/utils/cigar.py#L5), and which causes the license text to differ from the examples given in the [Ensembl docs on including the Apache 2.0 license in code](https://www.ebi.ac.uk/seqdb/confluence/display/ENS/Ensembl+Software+License#EnsemblSoftwareLicense-Apache2.0LicenseinCode). Please consider changing the license text to match the recommended text.

File: src/python/lib/ensembl/compara/utils/cigar.py [hunk 1]
```diff
@@ -14,116 +14,101 @@
...
+            result.append((op_length, c))
+            number = ""
     return result
 
 
 def alignment_to_seq_coordinate(cigar: str, coord: int) -> List[int]:
-    """Covert coordinate from the alignment level to sequence level
+    """Convert coordinate from the alignment level to sequence level
 
     This function will convert a position at the coordinate level to the sequence level. To be able to
```
[the remark this answers is not in the document]
Comment (dthybert) [reply src/python/lib/ensembl/compara/utils/cigar.py:90]: yes

File: src/python/tests/test_cigar.py [hunk 2]
```diff
@@ -48,11 +45,11 @@ class TestCigar:
         ],
     )
     def test_aligned_seq_to_cigar(self, aligned_seq: str, cigar: str) -> None:
-        """Tests :meth:`cigar.aligned_seq_to_cigar()` method.
+        """Tests :meth:`cigar.aligned_seq_to_cigar()` function.
```
Comment (twalsh-ebi) [inline src/python/tests/test_cigar.py:48]:
```suggestion
        """Tests :func:`cigar.aligned_seq_to_cigar()` function.
```

File: src/python/tests/test_cigar.py [hunk 3]
```diff
@@ -94,15 +89,15 @@ def test_get_cigar_array(self, cigar: str, cigar_array: List[Tuple]) -> None:
     )
     def test_alignment_to_seq_coordinate(self, cigar: str, coord_align: int,
                                          coord_seq: int, expectation: ContextManager) -> None:
-        """Tests :meth:`cigar.alignment_to_seq_coordinate()` method.
+        """Tests :func:`cigar.alignment_to_seq_coordinate()` method.
```
Comment (twalsh-ebi) [inline src/python/tests/test_cigar.py:92]:
```suggestion
        """Tests :func:`cigar.alignment_to_seq_coordinate()` function.
```

File: src/python/tests/test_cigar.py [hunk 4]
```diff
@@ -70,10 +67,10 @@ def test_get_cigar_array(self, cigar: str, cigar_array: List[Tuple]) -> None:
         """Tests :meth:`cigar.aligned_seq_to_cigar()` method.
```
Comment (twalsh-ebi) [inline src/python/tests/test_cigar.py:67]:
```suggestion
        """Tests :func:`cigar.aligned_seq_to_cigar()` function.
```

**Review by twalsh-ebi — COMMENTED** [review]
It's looking grand now. I just have one suggestion (in three comments) to be unambiguous about the members in the `cigar` module being functions rather than methods.

**Pull request merged**